include/rocksdb/slice_transform.h

```diff
@@ -47,8 +47,11 @@ class SliceTransform : public Customizable {
                                  std::shared_ptr<const SliceTransform>* result);
 
   // Returns a string representation of this SliceTransform, representing the ID
-  // and any additional properties
-  std::string AsString() const;
+  // and any additional properties.
+  // The default implementation is optimized for instances which have no
+  // additional properties/options.  Instances which have properties/options
+  // should override this method and call ToString();
```

Comment:
I would prefer not to involve the API extension user in performance concerns in non-obvious ways.

(See #9407)